src/config.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::registry::RegistryUri;
+use miette::{bail, ensure, miette, Context, IntoDiagnostic};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+/// Representation of the .config/buffrs/config.toml configuration file
+///
+/// # Example
+///
+/// ```toml
+/// [registries]
+/// some_org = "https://artifactory.example.com/artifactory/some-org"
+///
+/// [registry]
+/// default = "some_org"
+/// ```
+///
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Config {
+    /// Path to the configuration file
+    config_path: Option<PathBuf>,
+
+    /// Default registry to use if none is specified
+    default_registry: Option<String>,
+
+    /// List of registries
+    registries: HashMap<String, RegistryUri>,
+}
+
+impl Config {
+    /// Create a new configuration with default values
+    /// # Arguments
+    /// * `cwd` - Starting directory to search for the configuration file
+    ///
+    pub fn new(cwd: Option<&Path>) -> miette::Result<Self> {
+        match Self::locate_config(cwd) {
+            Some(config_path) => Self::new_from_config_file(&config_path),
+            None => Ok(Self {
+                config_path: None,
+                default_registry: None,
+                registries: HashMap::new(),
+            }),
+        }
+    }
```

Comment:
```suggestion
    pub fn from_config() -> miette::Result<Self> {
        let config_path = etcetera::choose_base_strategy()?.config_dir().join("buffrs/config.toml");
        if config_path.exists() {
            Some(config_path) => Self::new_from_config_file(&config_path),
            None => Ok(Self {
                config_path: None,
                default_registry: None,
                registries: HashMap::new(),
            }),
        }
    }
```

Could probably be simplified further, not sure we need both `new_from_config_file` and `from_config`.